packages/esm-bed-management-app/src/bed-tag/edit-tag-form.modal.tsx

```diff
@@ -0,0 +1,191 @@
+
+interface EditBedTagFormProps {
+  editData: BedTagData;
+  mutate: Mutator<BedTagData>;
+  closeModal: () => void;
+}
+
+const BedTagAdministrationSchema = z.object({
+  name: z.string().max(255),
+});
+
+interface ErrorType {
+  message: string;
+}
+
+const EditBedTagForm: React.FC<EditBedTagFormProps> = ({ editData, mutate, closeModal }) => {
+  const { t } = useTranslation();
+  const [{ isErrored, isSubmitting, isSuccessful, description }, setSubmissionStatus] = useState<{
+    isSubmitting: boolean;
+    isSuccessful: boolean | undefined;
+    isErrored: boolean | undefined;
+    description: string | undefined;
+  }>({
+    isSubmitting: false,
+    isSuccessful: false,
+    isErrored: undefined,
+    description: undefined,
+  });
+
+  const handleUpdateBedTag = useCallback(
+    (formData: BedTagDataAdministration) => {
+      const bedUuid = editData.uuid;
+      const { name } = formData;
+
+      const bedTagPayload = {
+        name,
+      };
+
+      setSubmissionStatus({
+        isErrored: false,
+        isSuccessful: undefined,
+        isSubmitting: true,
+        description: 'Submitting...',
+      });
+      editBedTag({ bedTagPayload, bedTagId: bedUuid })
+        .then(() => {
+          showSnackbar({
+            kind: 'success',
+            title: t('bedTagUpdated', 'Bed tag updated'),
+            subtitle: t('bedTagUpdatedSuccessfully', `${bedTagPayload.name} updated successfully`, {
+              bedTag: bedTagPayload.name,
+            }),
+          });
+          mutate();
+          setSubmissionStatus({
+            isErrored: false,
+            isSuccessful: true,
+            isSubmitting: false,
+            description: 'Saved...',
+          });
+        })
+        .catch((error) => {
+          showSnackbar({
+            kind: 'error',
+            title: t('errorCreatingBedTag', 'Error creating bed tag'),
+            subtitle: error?.message,
+          });
+          setSubmissionStatus({
+            isErrored: true,
+            isSuccessful: false,
+            isSubmitting: false,
+            description: 'Errored',
+          });
+        })
+        .finally(() => {
+          closeModal();
+        });
+    },
+    [mutate, editData, t],
+  );
+
+  const [showErrorNotification, setShowErrorNotification] = useState(false);
+  const [formStateError, setFormStateError] = useState('');
+
+  const {
+    handleSubmit,
+    control,
+    formState: { isDirty },
+  } = useForm<BedTagData>({
+    mode: 'all',
+    resolver: zodResolver(BedTagAdministrationSchema),
+    defaultValues: {
+      name: editData.name || '',
+    },
+  });
+
+  const onSubmit = (formData: BedTagData) => {
+    const result = BedTagAdministrationSchema.safeParse(formData);
+    if (result.success) {
+      setShowErrorNotification(false);
+      handleUpdateBedTag(formData);
+    }
+  };
+
+  const onError = (error: { [key: string]: ErrorType }) => {
+    setFormStateError(Object.entries(error)[0][1].message);
```

Comment:
So we only support a single error? That seems wrong...

In principle, each field should display an error message related to that field and we only need a bigger message if there's an error... bigger than a field.

Reply:
@ibacher the aim of this PR was to just separate out the forms. As I said in the PR description I consciously made the decision to leave everything as is to make it easier for review. If more changes like this are required I am more than happy to work on them but that might bloat the PR and make it more challenging to review.

Also as I was working on this I made the assumption that since I was working on top of already merged changes there is not much need to concentrate on the details.

Reply:
Because of the way these changes are done, it's very hard for me to tell what's "new" in this PR and what's just "moved" from somewhere else.

Reply:
I see. Think of it this way. the `new tag form` and `the edit tag form` both used an abstracted `tag component`. what I did was duplicate the tag component in both forms. So the `new tag form` know owns its own functionality that was previously in the `tag component` same as the edit one.

Reply:
so nothing is new in short